pages/docs/reference/mpp-publish-lib.md

```diff
@@ -0,0 +1,195 @@
+
+You can detect the platform in the script, introduce a flag such as `isMainHost` and set it to `true` for the main target 
+platform. Alternatively, you can pass the flag from an external source, for example, from CI configuration. 
+
+This simplified example ensures that publications are only uploaded when `isMainHost=true` is passed. This means that 
+a publication that can be published from multiple platforms will be published only once – from the main host.
+
+<div class="multi-language-sample" data-lang="groovy">
+<div class="sample" markdown="1" theme="idea" mode="groovy" data-highlight-only>
+
+```groovy
+kotlin {
+    jvm()
+    js()
+    mingwX64()
+    linuxX64()
+
+    // Note that the Kotlin metadata is here, too.
+
+    configure([targets["metadata"], jvm(), js()]) {
+        mavenPublication { targetPublication ->
+            tasks.withType(AbstractPublishToMaven)
+                    .matching { it.publication == targetPublication }
+                    .all { onlyIf { findProperty("isMainHost") == "true" } }
+        }
+    }
+}
+```
+
+</div>
+</div>
+
+<div class="multi-language-sample" data-lang="kotlin">
+<div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+
+```kotlin
+kotlin {
+    jvm()
+    js()
+    mingwX64()
+    linuxX64()
+
+    // Note that the Kotlin metadata is here, too.
+
+    configure(listOf(targets["metadata"], jvm(), js())) {
+        mavenPublication {
+            val targetPublication = this@mavenPublication
+            tasks.withType<AbstractPublishToMaven>()
+                    .matching { it.publication == targetPublication }
+                    .all { onlyIf { findProperty("isMainHost") == "true" } }
+        }
+    }
+}
+```
+
+</div>
+</div>
+ 
+By default, each publication includes a sources JAR that contains the sources used by the main compilation of the target. 
+
+## Publish an Android library
+
+To publish an Android library, you need to provide additional configuration.
+
+By default, no artifacts of an Android library are published. To publish artifacts produced by a set of [Android variants](https://developer.android.com/studio/build/build-variants), 
+specify the variant names in the Android target block:
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+kotlin {
+    android {
+        publishLibraryVariants("release", "debug")
+    }
+}
+
+```
+
+</div>
+
+The example works for Android libraries without [product flavors](https://developer.android.com/studio/build/build-variants#product-flavors). 
+For a library with product flavors, the variant names also contain the flavors, like `fooBarDebug` or `fooBazRelease`.
+
+> If a library consumer defines variants that are missing in the library, they need to provide matching fallbacks. 
+> For example, if a library does not have or does not publish a staging build type, you must provide a fallback for the 
```

Comment:
Not 'you', the consumer must provide a matching fallback on their side. The library author doesn't know which fallbacks the consumer needs.

Reply:
🛠